packages/core/src/indexing/index.ts

```diff
@@ -57,12 +65,69 @@ export type Indexing = {
   }) => Promise<void>;
 };
 
+export type ColumnAccessProfile = {
+  blockInclude: typeof defaultBlockInclude;
+  traceInclude: typeof defaultTraceInclude;
+  transactionInclude: typeof defaultTransactionInclude;
+  transactionReceiptInclude: typeof defaultTransactionReceiptInclude;
+  defaultInclude: Set<string>;
+  accessed: Set<string>;
+  eventCount: number;
+  resolved: boolean;
+  resolve: () => void;
+};
+
+export const createColumnAccessProfile = ({
+  common,
+}: { common: Common }): ColumnAccessProfile => ({
+  blockInclude: [...defaultBlockInclude],
+  traceInclude: [...defaultTraceInclude],
+  transactionInclude: [...defaultTransactionInclude],
+  transactionReceiptInclude: [...defaultTransactionReceiptInclude],
+  accessed: new Set(),
```

Comment:
Do you think we should have the column selection be filter-specific for other data sources?